src/rard/research/tests/views/test_mentions.py

```diff
@@ -64,131 +116,263 @@ def test_default_queryset(self):
-
-        # fragments
-        f1 = Fragment.objects.create()
-        FragmentLink.objects.create(fragment=f1, antiquarian=a1)
-
-        f2 = Fragment.objects.create()
-        FragmentLink.objects.create(fragment=f2, antiquarian=a2)
-
-        self.assertEqual(list(view.fragment_search("findme")), [f1])
-        self.assertEqual(list(view.fragment_search("FINDM")), [f1])
-        self.assertEqual(list(view.fragment_search("foo")), [f2])
-        self.assertEqual(list(view.fragment_search("Fo")), [f2])
-        self.assertEqual(list(view.fragment_search("f")), [f1, f2])
-
-        t1 = Testimonium.objects.create()
-        TestimoniumLink.objects.create(testimonium=t1, antiquarian=a1)
-
-        t2 = Testimonium.objects.create()
-        TestimoniumLink.objects.create(testimonium=t2, antiquarian=a2)
-
-        self.assertEqual(list(view.testimonium_search("findme")), [t1])
-        self.assertEqual(list(view.testimonium_search("FIN")), [t1])
-        self.assertEqual(list(view.testimonium_search("foo")), [t2])
-        self.assertEqual(list(view.testimonium_search("fO")), [t2])
-        self.assertEqual(list(view.testimonium_search("F")), [t1, t2])
-
-        # anonymous fragments
-        af1 = AnonymousFragment.objects.create()
-        AppositumFragmentLink.objects.create(anonymous_fragment=af1, antiquarian=a1)
-        af2 = AnonymousFragment.objects.create()
-        AppositumFragmentLink.objects.create(anonymous_fragment=af2, antiquarian=a2)
-
-        self.assertEqual(list(view.anonymous_fragment_search("f1")), [af1])
-        self.assertEqual(list(view.anonymous_fragment_search("F1")), [af1])
-        self.assertEqual(list(view.anonymous_fragment_search("f2")), [af2])
-        self.assertEqual(list(view.anonymous_fragment_search("F 2")), [af2])
-        self.assertEqual(list(view.anonymous_fragment_search("F")), [af1, af2])
-        self.assertEqual(list(view.anonymous_fragment_search("")), [])
-        self.assertEqual(list(view.anonymous_fragment_search("1")), [])
-
-        # bibliography items
-        bi1 = BibliographyItem.objects.create(
-            authors="fee fie foe",
-            author_surnames="Froe",
-            title="apples and pears",
-            parent=a1,
+        # no search term
+        self.assertEqual(len(list(view.get_queryset())), 3)
+
+        # number as search: only applicable if number in authors/name
+        view.request = self.request(
+            data={
+                "q": "wk:2",
+            }
         )
-        bi2 = BibliographyItem.objects.create(
-            authors="tee tie toe",
-            author_surnames="Twoe",
-            title="peppers and carrots",
-            parent=a2,
+        self.assertEqual(len(list(view.get_queryset())), 0)
+
+        # string as search
+        view.request = self.request(
+            data={
+                "q": "wk:ro",
+            }
+        )
+        self.assertEqual(len(list(view.get_queryset())), 3)
```

Comment:
I don't think the number search is necessary here, but what's missing is a test to see if we're matching work authors' names. If you add works to antiquarians in the setup function we can test that here; e.g. "and" should match antiquarian "Andrew" and work "andropov"